src/logging/mod.rs

```diff
@@ -0,0 +1,162 @@
+                .map(&str::to_owned)
+                .or(request.remote().map(|addr| addr.ip().to_string())),
+        }
+    }
+}
+
+impl KV for RequestMozlogFields {
+    fn serialize(&self, _: &Record, serializer: &mut Serializer) -> slog::Result {
+        if let Some(ref agent) = self.agent {
+            serializer.emit_str("agent", agent)?;
+        }
+        if let Some(ref remote) = self.remote {
+            serializer.emit_str("remoteAddressChain", remote)?;
+        }
+        serializer.emit_str("path", &self.path)?;
+        serializer.emit_str("method", self.method)?;
+
+        Ok(())
+    }
+}
+
+impl Value for Settings {
+    fn serialize(&self, _: &Record, _: Key, serializer: &mut Serializer) -> slog::Result {
+        serializer.emit_str("auth_db_uri", &self.authdb.baseuri)?;
+        serializer.emit_str("provider", &self.provider)?;
+
+        if let Some(_) = self.aws.keys {
+            serializer.emit_str("aws_keys_access", "[hidden]")?;
+        } else {
+            serializer.emit_str("aws_keys_access", "[not set]")?;
+        }
+
+        serializer.emit_str("aws_region", &self.aws.region)?;
+
+        if let Some(ref sqsurls) = self.aws.sqsurls {
+            serializer.emit_str("aws_sqs_urls_bounce", &sqsurls.bounce.to_owned())?;
+            serializer.emit_str("aws_sqs_urls_complaint", &sqsurls.complaint.to_owned())?;
+            serializer.emit_str("aws_sqs_urls_delivery", &sqsurls.delivery.to_owned())?;
+            serializer.emit_str("aws_sqs_urls_notification", &sqsurls.notification.to_owned())?;
+        } else {
+            serializer.emit_str("aws_sqs_urls", "[not set]")?;
+        }
+
+        if let Some(ref jsonlogging) = self.jsonlogging {
+            serializer.emit_str("json_logging", &format!("{}", jsonlogging))?;
+        }
+
+        if let Some(_) = self.sendgrid {
+            serializer.emit_str("sendgrid_key", "[hidden]")?;
+        } else {
+            serializer.emit_str("sendgrid_key", "[not set]")?;
+        }
+
+        serializer.emit_str("smtp_host", &self.smtp.host)?;
+        serializer.emit_str("smtp_port", &format!("{}", self.smtp.port))?;
+        
+        if let Some(ref user) = self.smtp.user {
+            serializer.emit_str("smtp_user", user)?;
+        } else {
+            serializer.emit_str("smtp_user", "[not set]")?;
+        }
+
+        if let Some(_) = self.smtp.password {
+            serializer.emit_str("smtp_password", "[hidden]")?;
+        } else {
+            serializer.emit_str("smtp_password", "[not set]")?;
+        }
+
+        serializer.emit_str("bounce_limits_enabled",  &format!("{}", &self.bouncelimits.enabled))?;
+        serializer.emit_str("bounce_limits_complaint",  &format!("{:?}", &self.bouncelimits.complaint))?;
+        serializer.emit_str("bounce_limits_hard",  &format!("{:?}", &self.bouncelimits.hard))?;
+        serializer.emit_str("bounce_limits_soft",  &format!("{:?}", &self.bouncelimits.soft))?;
```

Comment:
I really like this function but it's pretty verbose. This will do for now but longer-term we should think about how we might be able to eradicate some of the repeated boilerplate code here.

Like would it be possible to have two static lists that define the sensitive and non-sensitive properties and just iterate through the lists? Or would it be possible to annotate them with a macro in `src/settings/mod.rs` so that we can determine whether or not they're sensitive programmatically?

Anything that makes it less likely for us to accidentally leak sensitive data to the logs is worth thinking about.

Reply:
I agree. I'll open an issue for that after this PR goes through :)